src/query/allowed_objects.c

```diff
@@ -14,18 +14,49 @@ static const char *const g_allowed_casts[] = {
     "ftod", "dtof",
     "int4_numeric", "float4_numeric", "float8_numeric",
     "numeric_float4", "numeric_float8",
+    "date_timestamptz",
     /**/
 };
 
-static const char *const g_allowed_builtins[] = {
+typedef struct FunctionWithName
```

Comment:
Maybe drop the `With`?

Reply:
hm, that would make these types `FunctionName` and `FunctionOid`. Could work, but I've had a late thought: `FunctionByName` and `FunctionByOid` - what we actually are achieving here is a mapping, just with linear lookup. Would these work?

And a more general question, to make sure - I'm assuming the :heavy_check_mark: means you prefer the current version with the arrays to the alternative with a runtime-populated dictionary?

Reply:
`FunctionByName` and `FunctionByOid` sound good to me.

For the small collections we have, I think linear searches are fine. If we ever find they are a bottleneck, we can pre-sort the arrays and do a binary search instead, to get a speed boost.

Reply:
to clarify - my focus here is more on code clarity, not the speed of lookups. But if that's fine by you as well, let's keep the current approach.

Reply:
This being C, I think the array version is the clearer one :)